src/uct/ib/rdmacm/rdmacm_ep.h

```diff
@@ -9,20 +9,28 @@
 #include "rdmacm_iface.h"
 
 struct uct_rdmacm_ep {
-    uct_base_ep_t                      super;
-    void                               *priv_data;
-    ucs_list_link_t                    list_elem;       /* for the pending_eps_list*/
-    struct sockaddr_storage            remote_addr;
-    int                                is_on_pending;
-    uct_worker_cb_id_t                 slow_prog_id;
-    uct_rdmacm_ctx_t                   *cm_id_ctx;
+    uct_base_ep_t                          super;
+    uct_sockaddr_fill_priv_data_callback_t fill_priv_data_cb;
+    void                                   *fill_priv_data_arg;
+    uint32_t                               cb_flags;
+    ucs_list_link_t                        list_elem;   /* for the pending_eps_list */
+    struct sockaddr_storage                remote_addr;
+    int                                    is_on_pending;
+    uct_worker_cb_id_t                     slow_prog_id;
+    uct_rdmacm_ctx_t                       *cm_id_ctx;
 };
 
+typedef struct uct_rdmacm_ep_err_handle_cb_arg {
+    uct_rdmacm_ep_t *rdmacm_ep;
+    ucs_status_t    status;
```

Comment:
maybe save the status on the ep to avoid having another struct

Reply:
this status is used only for the error handling flow and only for a SYNC case. why have it always on the ep if it's required so rarely?

Reply:
yes but status is small, rdma_cm ep is short lived, so memory consumption is not issue, but on other hand it will reduce code